### ISSUE TITLE
Numerical INstability

### ISSUE THREAD

**Issue by hywkim-brian** [oss-roettger/T5-Textual-Inversion#1]
Hey I have. re-implemented a part of your code to suite my research purposes.. Did you not experience any issues with gradient collapse/explosion (ie grad being set to nan/inf) while training ? If so, how did you work around this ?? Many thanks !

**Comment by oss-roettger** [comment]
Hey hywkim, no I have not experienced any gradient collapse/explosion/NaN/inf so far. Please note that I have used the original FP32 models, optimized with AdamW, applied gradient accumulation (equal to batch size=4) and scaled the loss by a factor of 1000 (diffusion.num_timesteps). Hope this information helps you solving your problems.

**Comment by hywkim-brian** [comment]
Thanks so much! I think using FP32 instead of FP16 is crucial for finetuning the model without gradient collapse. though sadly it eats a lot of memory so I can't train on negative prompts anymore..
